drivers/modem/ublox-sara-r4.c

```diff
@@ -1216,15 +1216,13 @@ static int offload_close(void *obj)
 		return 0;
 	}
 
-	if (sock->is_connected || sock->ip_proto == IPPROTO_UDP) {
-		snprintk(buf, sizeof(buf), "AT+USOCL=%d", sock->id);
+	snprintk(buf, sizeof(buf), "AT+USOCL=%d", sock->id);
 
-		ret = modem_cmd_send(&mctx.iface, &mctx.cmd_handler,
-				     NULL, 0U, buf,
-				     &mdata.sem_response, MDM_CMD_TIMEOUT);
-		if (ret < 0) {
-			LOG_ERR("%s ret:%d", log_strdup(buf), ret);
-		}
+	ret = modem_cmd_send(&mctx.iface, &mctx.cmd_handler,
+			     NULL, 0U, buf,
+			     &mdata.sem_response, MDM_CMD_TIMEOUT);
+	if (ret < 0) {
+		LOG_ERR("%s ret:%d", log_strdup(buf), ret);
```

Comment:
Perhaps we should make this a LOG_WARN now?  I feel like users may see this error more often if we do an unconditional +USOCL.

Reply:
I'm okay with that change. One error condition being prevented here was issuing a +USOCL for a socket which was already closed by the remote peer. This wouldn't be unusual to encounter in a ported POSIX application so a lower log level sounds good to me.

We could also change the code to have something like an `is_allocated` or `is_valid` flag so we can catch that case but still allow sending +USOCL on an unconnected socket.

Reply:
I think that the log should not be displayed when we already now that the socket has been closed, and that we made the call only to avoid a leak in sockets in the modem internals.